lib/mongo/scrollable.rb

```diff
@@ -28,6 +29,19 @@ def scroll(cursor = nil, options = nil, &_block)
         view
       end
     end
+
+    private
+
+    def raise_mismatched_sort_fields_error!(cursor, criteria_cursor_options)
```

Comment:
`raise_mismatched_sort_fields_error!` and `different_sort_fields` are duplicated in each `*::Scrollable` module.

It could be extracted to a module which would be included by the different `*::Scrollable`  modules to reduce duplication. Tell me if you want me to do it, and if so, in which file this new module should be created.

Reply:
Makes sense. Pick a location, I'll think a bit more where it belongs.

Reply:
Maybe `lib/mongoid/criteria/scrollable/fields`?